carbonmark-data/app/[locale]/retirement-trends/page.tsx

```diff
@@ -0,0 +1,186 @@
+          {children}
+        </TabPanel>
+      );
+  }
+  return assertUnreachable(tab);
+};
+
+export default function RetirementTrends() {
+  const [activeTab, setActiveTab] = useState("byPool");
+
+  const handleChange = (_: React.SyntheticEvent, newTab: string) => {
+    setActiveTab(newTab);
+  };
+
+  return (
+    <>
+      <h1>{t`Retirement Trends`}</h1>
+      <TabContext value={activeTab}>
+        <MobileTabSelector<Tab>
+          value={activeTab}
+          options={tabs.map((tab) => {
+            switch (tab) {
+              case "byPool":
+                return { label: t`By Pool`, value: tab };
+              case "byToken":
+                return { label: t`By Token`, value: tab };
+              case "byChain":
+                return { label: t`By Chain`, value: tab };
+              case "byBeneficiary":
+                return { label: t`By Beneficiary`, value: tab };
+            }
+            return assertUnreachable(tab);
+          })}
+          onSelectionChanged={(tab: string) => setActiveTab(tab)}
+          className={`${styles.mobileOnly} ${styles.mobileTabSelector}`}
+        />
+        <TabList
+          value={activeTab}
+          onChange={handleChange}
+          className={styles.desktopOnly}
+        >
+          <Tab
+            className={styles.tabButton}
+            label={t`By Pool`}
+            value="byPool"
+          ></Tab>
+          <Tab
+            className={styles.tabButton}
+            label={t`By Token`}
+            value="byToken"
+          ></Tab>
+          <Tab
+            className={styles.tabButton}
+            label={t`By Chain`}
+            value="byChain"
+          ></Tab>
+          <Tab
+            className={styles.tabButton}
+            label={t`By Beneficiary`}
+            value="byBeneficiary"
+          ></Tab>
+        </TabList>
+        <TypedTabPanel tab="byPool" className={styles.noPadding}>
+          <div className={layout.twoColumns}>
+            <div className={layout.cardStackedRows}>
+              <div className={layout.cardRow}>
+                <ChartCard
+                  isColumnCard={true}
+                  title="KlimaDAO retirements by pool"
+                />
+              </div>
+              <div className={layout.cardRow}>
+                <ChartCard
+                  isColumnCard={true}
+                  title="KlimaDAO retirements by pool"
+                />
+              </div>
+              <div className={layout.cardRow}>
+                <ChartCard
+                  isColumnCard={true}
+                  title="Detailed list of KlimaDAO retirements"
+                />
+              </div>
+            </div>
+            <div className={layout.cardStackedRows}>
+              <ChartCard
+                isColumnCard={true}
+                title="Carbon pool redemptions / retirements"
+              />
+            </div>
+          </div>
+        </TypedTabPanel>
```

Comment:
I think we should put each panel in separate file components for lisibility.

Reply:
done in [5d08010](https://github.com/KlimaDAO/klimadao/pull/1518/commits/5d080105445a3a09bcdb014bbb744393cdf730b0)

is it enough?

Reply:
Ha sorry! That's not what I meant, but do not worry though, we can do that when implementing the actual tab contents.